app/controllers/insights_cloud/settings_controller.rb

```diff
@@ -1,6 +1,12 @@
 module InsightsCloud
   class SettingsController < ::ApplicationController
     def show
+      if ForemanRhCloud.with_local_advisor_engine?
+        render json: {
+          error: _('This Foreman is configured to use Insights on Premise. Syncing to Insights Cloud is disabled.'),
+        }, status: :unprocessable_entity
+        return
+      end
       render_setting(:insightsSyncEnabled, :allow_auto_insights_sync)
```

Comment:
Nit: is this easier to follow? I generally try to avoid `return` and rely on the visual indentation to follow the code.

```suggestion
      else
        render_setting(:insightsSyncEnabled, :allow_auto_insights_sync)
      end
```

Though taking a step back, from an API perspective I wonder why this is even here in the first place. We already have an API to show settings: `/api/settings/:id` that supports both [GET](https://apidocs.theforeman.org/katello/latest/apidoc/v2/settings/show.html) and [PUT](https://apidocs.theforeman.org/katello/latest/apidoc/v2/settings/update.html).

Are there any permission checks on this controller to prevent unauthenticated users to change the setting or can any user change this for any organization?

Reply:
> Are there any permission checks on this controller to prevent unauthenticated users to change the setting or can any user change this for any organization?

It looks like settings endpoints need the `:generate_foreman_rh_cloud` permission:

https://github.com/theforeman/foreman_rh_cloud/blob/dc80828e3949cd9967121f0a251639561ec2124a/lib/foreman_rh_cloud/engine.rb#L63